### ISSUE TITLE
Problem in test_compute_comprehensiveness()

### ISSUE THREAD

**Issue by nc1m** [automl-classroom/iML-ws21-ex09#2]
Hi,
I'm having trouble getting compute_comprehensiveness()/test_compute_comprehensiveness() to work.
The following assertion fails:
>       assert torch.allclose(scores[:4], torch.tensor([-0.2626, -0.0613, 0.5092, 0.4640]), atol=1e-2)

The weird thing is the first two value of scores[:4] match exactly the expected values:
tensor([-0.2626, -0.0613,  0.3417,  0.3760])

Now I'm not sure if it's a seed problem or if I have an error in my code:
```
def get_predictions(model, X, feature_mask=None):
    X_masked = X.clone()
    if feature_mask is not None:
        X_masked[feature_mask == 0] = 0
    pred_labels, pred_probabilities = model.predict(X_masked)
    return pred_labels, pred_probabilities


def compute_comprehensiveness(model, X, features_to_remove=None):
    if features_to_remove is not None:
        mask = torch.ones_like(X)
        mask.scatter_(1, features_to_remove, 0)
    else:
        mask = None
    y_pred_labels, y_pred_probs = get_predictions(model, X)
    yPrime_pred_labels, yPrime_pred_probs = get_predictions(model, X, mask)
    y_pred_labels = y_pred_labels.unsqueeze(1)
    yPrime_pred_labels = yPrime_pred_labels.unsqueeze(1)
    comprehensiveness = torch.gather(y_pred_probs, 1, y_pred_labels) - torch.gather(yPrime_pred_probs, 1, yPrime_pred_labels)
    return comprehensiveness.squeeze(), torch.mean(comprehensiveness)
```

Greetings

**Comment by maxidl** [comment]
Hi,
when computing the difference in probability before and after removing features (y and yPrime), make sure to select the probabilities from the same class (the originally predicted class that is).
In your code, you could end up taking the difference of different class probabilities, when y_pred_labels and yPrime_pred_labels differ. If you select the probabilities using the original predictions y_pred_labels both times, your scores should match.

**Comment by nc1m** [comment]
Oh I see now, thank you!
